flipt-client-node/src/index.ts

```diff
@@ -1,202 +1,265 @@
+    this.fetcher = fetcher;
+  }
 
-const libfile = libFiles[key];
+  static async init(
+    namespace: string = 'default',
+    options: ClientOptions<AuthenticationStrategy> = {
+      url: 'http://localhost:8080',
+      reference: '',
+      updateInterval: 120
+    }
+  ): Promise<FliptEvaluationClient> {
+    let url = options.url ?? 'http://localhost:8080';
+    url = url.replace(/\/$/, '');
+    url = `${url}/internal/v1/evaluation/snapshot/namespace/${namespace}`;
 
-if (!libfile) {
-  throw new Error(`Unsupported platform: ${platform}/${arch}`);
-}
+    if (options.reference) {
+      url = `${url}?reference=${options.reference}`;
+    }
 
-// get absolute path to libfliptengine
-const engineLibPath = path.join(__dirname, '..', 'ext', libfile);
+    const headers = new Headers();
+    headers.append('Accept', 'application/json');
+    headers.append('x-flipt-accept-server-version', '1.47.0');
 
-const engineLib = ffi.Library(engineLibPath, {
-  initialize_engine: ['void *', ['char **', 'string']],
-  evaluate_variant: ['char **', ['void *', 'string']],
-  evaluate_boolean: ['char **', ['void *', 'string']],
-  evaluate_batch: ['char **', ['void *', 'string']],
-  list_flags: ['char **', ['void *']],
-  destroy_engine: ['void', ['void *']],
-  destroy_string: ['void', ['char **']]
-});
+    if (options.authentication) {
+      if ('clientToken' in options.authentication) {
+        headers.append(
+          'Authorization',
+          `Bearer ${options.authentication.clientToken}`
+        );
+      } else if ('jwtToken' in options.authentication) {
+        headers.append(
+          'Authorization',
+          `JWT ${options.authentication.jwtToken}`
+        );
+      }
+    }
 
-export class FliptEvaluationClient {
-  private engine: Pointer<unknown>;
-
-  /**
-   * Initializes a new instance of the engine with the specified options.
-   *
-   * @param namespace - An optional namespace string for the engine instance. If not provided,
-   * `default` namespace will be used.
-   *
-   * @param engine_opts - The optional options for configuring the engine. If not provided,
-   *                      default values will be used.
-   *
-   * @returns Flipt evaluation client
-   *
-   * @throws Error
-   * This exception is thrown if platform is not supported.
-   */
-  public constructor(
-    namespace?: string,
-    engine_opts: EngineOpts<AuthenticationStrategy> = {
-      url: 'http://localhost:8080',
-      update_interval: 120,
-      reference: ''
+    let fetcher = options.fetcher;
+
+    if (!fetcher) {
+      fetcher = async (opts?: { etag?: string }) => {
+        if (opts && opts.etag) {
+          headers.append('If-None-Match', opts.etag);
+        }
+
+        const resp = await fetch(url, {
+          method: 'GET',
+          headers
+        });
+
+        if (resp.status === 304) {
+          return resp;
+        }
+
+        if (!resp.ok) {
+          throw new Error('Failed to fetch data');
+        }
```

Comment:
Some minor thing but could we add more details about the status in the error.  It would so easier to understand the problem if we add some `statusText` into the error message.  Also I would like double check if we want accept 200-299 range. Probably only 200 should be valid status code, right?

Reply:
I added `statusText` to the errors where it can occur, im not to worked about non 2xx errors though